docs/source/Contribs/Components.md

```diff
@@ -0,0 +1,119 @@
+# Components
+
+_Contrib by ChrisLR 2021_
+
+# The Components Contrib
+
+This contrib allows you to develop your mud with more composition and less inheritance.
+
+Why would you want to do this?
+It allows you to keep your code clean and focused to one feature at a time.
+
+For example, a 'health' component which contains the current and max health
+but also the damage/heals method and various logic about dying.
+Then all you have to do is give this component to any object that can be damaged.
+After that you could filter any targets for the presence of the .health component
+
+if not target.health:
+    return "You cannot attack this!"
+else:
+    damage = roll_damage()
+    target.damage(damage)
+
+
+It also works similarly to attributes for components which are not declared
+at compile time, you can use something like
+
+if health := target.components.get('health'):
+    health.heal(10)
+
+
+# How to install
+
+To enable component support for a typeclass,
+import and inherit the ComponentHolderMixin, similar to this
+```
+from evennia.contrib.components import ComponentHolderMixin
+class Character(ComponentHolderMixin, DefaultCharacter):
+```
+
+Components need to inherit the Component class directly and require a name
+Example:
+```
+from evennia.contrib.components import Component
+
+class Health(Component):
+    name = "health"
+```
+
+Components may define DBFields or NDBFields at the class level
+Example:
+```
+from evennia.contrib.components import Component, DBField
+
+class Health(Component):
+    health = DBField(default=1)
+```
+
+Note that default is optional
+
+
+Each typeclass using the ComponentHolderMixin can declare its components
+in the class via the ComponentProperty.
+These are components that will always be present in a typeclass.
```

Comment:
It may be worth comparing these to how the `AttributeProperty` works in regular (develop) Evennia.